### ISSUE TITLE
Serve up social metadata so Facebook & friends (and search) show appropriate previews for experiments

### ISSUE THREAD

**Issue by lmorchard** [mozilla/testpilot#761]
Related to #419.

But, more than just adding views to our client-side app. This issue requires that we [add meta headers to our HTML on experiment URL routes on the server side so that Facebook's open graph spider picks them up](https://developers.facebook.com/docs/sharing/best-practices#tags).

**Comment by pdehaan** [comment]
via @edmoz:

![](http://people.mozilla.org/~ewong2/fb.png)

**Comment by ckprice** [comment]
Moving into this sprint.

**Comment by ckprice** [comment]
Quick note that this also effects SERPs (search engine results pages).

![SERP](https://s3.amazonaws.com/f.cl.ly/items/0T2R1S2s0b3W2X3T3N2b/Image%202016-05-10%20at%202.45.05%20PM.png?v=a06eafc8)

**Comment by ckprice** [comment]
> But, more than just adding views to our client-side app. This issue requires that we add meta headers to our HTML on experiment URL routes on the server side so that Facebook's open graph spider picks them up.

I'd like to bring down the scope of this to just simple meta tags (title, description, image) on [`/frontend/templates/testpilot/frontend/index.html`](https://github.com/mozilla/testpilot/blob/master/testpilot/frontend/templates/testpilot/frontend/index.html) if we can.

This would generate the same preview for all pages, yes, but until #419 is completed sharing out experiment URLs is going to be funky anyway. Not optimal, but right now we don't have anything coming up and I've received a couple of reports about this.

@lmorchard thoughts?

@johngruen can we get a preview image? [Facebook's sharing best practices](https://developers.facebook.com/docs/sharing/best-practices#tags) says

> Image Sizes
> Use images that are at least 1200 x 630 pixels for the best display on high resolution devices. At the minimum, you should use images that are 600 x 315 pixels to display link page posts with larger images. Images can be up to 8MB in size.

**Comment by lmorchard** [comment]
> I'd like to bring down the scope of this to just simple meta tags (title, description, image) on /frontend/templates/testpilot/frontend/index.html if we can.

Yeah, I think that would be a good first step, at least. Only need to change one spot, don't need to add per-experiment routes. 

We could iterate on adding those later and get _something_ better now even if it's the same for all URLs. That will be kind of a moot point until #419 is resolved, anyway.

**Comment by pdehaan** [comment]
I believe this was fixed by #922 (see https://github.com/mozilla/testpilot/issues/909#issuecomment-221712075 for screenshots)

I didn't verify SERPs, so I'll lazily reassign this to @ckprice to verify and close if he's happy (and file new issues if he's not -- we should probably file a new issue for experiment specific social metadata and set it as blocked by #419 regardless).

**Comment by ckprice** [comment]
Closing this. There are a couple of issues that #419 will spawn, we can file those when its ready to be worked on.

**Comment by lmorchard** [comment]
Going to re-open this, because I think it's what we're looking for in TXP-9 - but public detail pages (#419) are in [TXP-17](https://github.com/mozilla/testpilot/milestone/16)

**Comment by lmorchard** [comment]
Copying this over from #419:

Something hacky / simple we could do here so we have proper social sharing metadata along with public pages via Ampersand (#419): 
- Check for experiment URL patterns in [the catch-all frontend view](https://github.com/mozilla/testpilot/blob/master/testpilot/frontend/views.py)
- Fetch the Experiment model & pass it to the template.
- Render the appropriate per-experiment meta-tags in `<head>` in [the frontend template](https://github.com/mozilla/testpilot/blob/master/testpilot/frontend/templates/testpilot/frontend/index.html#L3) instead of the site-generic info

That way, we don't necessarily need to render complete pages for experiments on the server-side, or add all the server-side URL routes to accommodate them.

**Comment by chuckharmston** [comment]
@johngruen Adding you as an assignee, as we'll need content for this when shipping.
